File: src/lib/utils/utils.pyodidie/python/utils.pyodide.file-ops.py
def normalize_path(path: str) -> str:
    """Normalize path by removing ./ if present."""
    return path.replace("./", "", 1) if path.startswith("./") else path


def read_file(path: str) -> str:
    """Read a file's contents."""
    path = normalize_path(path)
    with open(path, "r") as f:
        return f.read()


def write_file(path: str, content: str) -> None:
    """Write content to a file."""
    path = normalize_path(path)
    with open(path, "w") as f:
        f.write(content)


def verify_file(path: str) -> None:
    """Verify a file can be read."""
    with open(path, "r") as f:
        f.read()

File: src/lib/utils/utils.pyodidie/utils.pyodide.file-ops.ts
import type { PyodideInterface } from './utils.pyodide.types';

export async function readFile(
	pyodide: PyodideInterface,
	path: string
): Promise<string> {
	return await pyodide.runPythonAsync(`read_file('${path}')`);
}

export async function writeFile(
	pyodide: PyodideInterface,
	path: string,
	content: string
): Promise<void> {
	const cleanContent = content
		.replace(/\\/g, '\\\\')
		.replace(/'''/g, "\\'\\'\\'");

	await pyodide.runPythonAsync(
		`write_file('${path}', ${JSON.stringify(cleanContent)})`
	);
}

export async function executeFile(
	pyodide: PyodideInterface,
	path: string
): Promise<void> {
	await pyodide.runPythonAsync(`execute_file('${path}')`);
}
